src/algebra/big_operators/order.lean

```diff
@@ -308,3 +309,57 @@ open opposite
 (@unop_add_hom β _).map_sum _ _
 
 end with_top
+
+namespace finset
+
+/--
+The strong pigeonhole principle for finitely many pigeons and pigeonholes.
+-/
+lemma strong_pigeonhole {s : finset α} {t : finset β} (hne : t.nonempty) [decidable_eq β]
```

Comment:
The lemma is true even without `hne`, isn't it?

Reply:
If both `s` and `t` are empty, then the remaining hypotheses can be trivially satisfied, but the conclusion would imply `t` is nonempty, a contradiction.  Putting a constraint on `t` being nonempty is enough to prevent this situation from happening.

Reply:
Hmm, I suppose you end up needing to prove `∃ y ∈ t, true`, which requires `t` to be non-empty.